File: tests/algorithms/x/Java/maths/series/harmonic.java
public class Main {

    static boolean is_harmonic_series(double[] series) {
        if (series.length == 0) {
            throw new RuntimeException(String.valueOf("Input list must be a non empty list"));
        }
        if (series.length == 1) {
            if (series[0] == 0.0) {
                throw new RuntimeException(String.valueOf("Input series cannot have 0 as an element"));
            }
            return true;
        }
        double[] rec_series = ((double[])(new double[]{}));
        int i = 0;
        while (i < series.length) {
            double val = series[i];
            if (val == 0.0) {
                throw new RuntimeException(String.valueOf("Input series cannot have 0 as an element"));
            }
            rec_series = ((double[])(java.util.stream.DoubleStream.concat(java.util.Arrays.stream(rec_series), java.util.stream.DoubleStream.of(1.0 / val)).toArray()));
            i = i + 1;
        }
        double common_diff = rec_series[1] - rec_series[0];
        int idx = 2;
        while (idx < rec_series.length) {
            if (rec_series[idx] - rec_series[idx - 1] != common_diff) {
                return false;
            }
            idx = idx + 1;
        }
        return true;
    }

    static double harmonic_mean(double[] series) {
        if (series.length == 0) {
            throw new RuntimeException(String.valueOf("Input list must be a non empty list"));
        }
        double total = 0.0;
        int i_1 = 0;
        while (i_1 < series.length) {
            total = total + 1.0 / series[i_1];
            i_1 = i_1 + 1;
        }
        return (((Number)(series.length)).doubleValue()) / total;
    }
    public static void main(String[] args) {
        {
            long _benchStart = _now();
            long _benchMem = _mem();
            System.out.println(is_harmonic_series(((double[])(new double[]{1.0, 2.0 / 3.0, 1.0 / 2.0, 2.0 / 5.0, 1.0 / 3.0}))));
            System.out.println(is_harmonic_series(((double[])(new double[]{1.0, 2.0 / 3.0, 2.0 / 5.0, 1.0 / 3.0}))));
            System.out.println(harmonic_mean(((double[])(new double[]{1.0, 4.0, 4.0}))));
            System.out.println(harmonic_mean(((double[])(new double[]{3.0, 6.0, 9.0, 12.0}))));
            long _benchDuration = _now() - _benchStart;
            long _benchMemory = _mem() - _benchMem;
            System.out.println("{");
            System.out.println("  \"duration_us\": " + _benchDuration + ",");
            System.out.println("  \"memory_bytes\": " + _benchMemory + ",");
            System.out.println("  \"name\": \"main\"");
            System.out.println("}");
            return;
        }
    }

    static boolean _nowSeeded = false;
    static int _nowSeed;
    static int _now() {
        if (!_nowSeeded) {
            String s = System.getenv("MOCHI_NOW_SEED");
            if (s != null && !s.isEmpty()) {
                try { _nowSeed = Integer.parseInt(s); _nowSeeded = true; } catch (Exception e) {}
            }
        }
        if (_nowSeeded) {
            _nowSeed = (int)((_nowSeed * 1664525L + 1013904223) % 2147483647);
            return _nowSeed;
        }
        return (int)(System.nanoTime() / 1000);
    }

    static long _mem() {
        Runtime rt = Runtime.getRuntime();
        rt.gc();
        return rt.totalMemory() - rt.freeMemory();
    }
}

File: tests/algorithms/transpiler/FS/maths/series/harmonic.fs
// Generated 2025-08-08 18:09 +0700

exception Return
let mutable _nowSeed:int64 = 0L
let mutable _nowSeeded = false
let _initNow () =
    let s = System.Environment.GetEnvironmentVariable("MOCHI_NOW_SEED")
    if System.String.IsNullOrEmpty(s) |> not then
        match System.Int32.TryParse(s) with
        | true, v ->
            _nowSeed <- int64 v
            _nowSeeded <- true
        | _ -> ()
let _now () =
    if _nowSeeded then
        _nowSeed <- (_nowSeed * 1664525L + 1013904223L) % 2147483647L
        int _nowSeed
    else
        int (System.DateTime.UtcNow.Ticks % 2147483647L)

_initNow()
let _dictAdd<'K,'V when 'K : equality> (d:System.Collections.Generic.IDictionary<'K,'V>) (k:'K) (v:'V) =
    d.[k] <- v
    d
let _dictCreate<'K,'V when 'K : equality> (pairs:('K * 'V) list) : System.Collections.Generic.IDictionary<'K,'V> =
    let d = System.Collections.Generic.Dictionary<'K, 'V>()
    for (k, v) in pairs do
        d.[k] <- v
    upcast d
let _dictGet<'K,'V when 'K : equality> (d:System.Collections.Generic.IDictionary<'K,'V>) (k:'K) : 'V =
    match d.TryGetValue(k) with
    | true, v -> v
    | _ -> Unchecked.defaultof<'V>
let _idx (arr:'a array) (i:int) : 'a =
    if not (obj.ReferenceEquals(arr, null)) && i >= 0 && i < arr.Length then arr.[i] else Unchecked.defaultof<'a>
let __bench_start = _now()
let __mem_start = System.GC.GetTotalMemory(true)
let rec is_harmonic_series (series: float array) =
    let mutable __ret : bool = Unchecked.defaultof<bool>
    let mutable series = series
    try
        if (Seq.length (series)) = 0 then
            failwith ("Input list must be a non empty list")
        if (Seq.length (series)) = 1 then
            if (_idx series (0)) = 0.0 then
                failwith ("Input series cannot have 0 as an element")
            __ret <- true
            raise Return
        let mutable rec_series: float array = Array.empty<float>
        let mutable i: int = 0
        while i < (Seq.length (series)) do
            let ``val``: float = _idx series (i)
            if ``val`` = 0.0 then
                failwith ("Input series cannot have 0 as an element")
            rec_series <- Array.append rec_series [|(1.0 / ``val``)|]
            i <- i + 1
        let common_diff: float = (_idx rec_series (1)) - (_idx rec_series (0))
        let mutable idx: int = 2
        while idx < (Seq.length (rec_series)) do
            if ((_idx rec_series (idx)) - (_idx rec_series (idx - 1))) <> common_diff then
                __ret <- false
                raise Return
            idx <- idx + 1
        __ret <- true
        raise Return
        __ret
    with
        | Return -> __ret
let rec harmonic_mean (series: float array) =
    let mutable __ret : float = Unchecked.defaultof<float>
    let mutable series = series
    try
        if (Seq.length (series)) = 0 then
            failwith ("Input list must be a non empty list")
        let mutable total: float = 0.0
        let mutable i: int = 0
        while i < (Seq.length (series)) do
            total <- total + (1.0 / (_idx series (i)))
            i <- i + 1
        __ret <- (float (Seq.length (series))) / total
        raise Return
        __ret
    with
        | Return -> __ret
printfn "%b" (is_harmonic_series (unbox<float array> [|1.0; 2.0 / 3.0; 1.0 / 2.0; 2.0 / 5.0; 1.0 / 3.0|]))
printfn "%b" (is_harmonic_series (unbox<float array> [|1.0; 2.0 / 3.0; 2.0 / 5.0; 1.0 / 3.0|]))
printfn "%g" (harmonic_mean (unbox<float array> [|1.0; 4.0; 4.0|]))
printfn "%g" (harmonic_mean (unbox<float array> [|3.0; 6.0; 9.0; 12.0|]))
let __bench_end = _now()
let __mem_end = System.GC.GetTotalMemory(true)
printfn "{\n  \"duration_us\": %d,\n  \"memory_bytes\": %d,\n  \"name\": \"main\"\n}" ((__bench_end - __bench_start) / 1000) (__mem_end - __mem_start)
